src/registered.rs

```diff
@@ -65,30 +126,39 @@ impl SecretKey {
     /// Derives a key for a registered application protocol at the given path from the
     /// given seed. Each path element may consist of an index and (possibly empty) tag.
     ///
-    /// - `zip_number` is the number of the ZIP defining the application protocol.
-    ///   The corresponding hardened index (with empty tag) will be prepended to the
-    ///   `subpath` to obtain the full derivation path.
-    /// - `context_string` is an identifier for the context in which this key will be
-    ///   used. It must be globally unique.
-    ///
-    /// # Panics
-    ///
-    /// Panics if:
-    /// - the context string is empty or longer than 252 bytes.
-    /// - the seed is shorter than 32 bytes or longer than 252 bytes.
+    /// - `context_string`: an identifier for the context in which this key will be used. It must
+    ///   be globally unique. Must be no more than 252 bytes in length.
```

Comment:
This is changing the spec by allowing the empty context string. https://zips.z.cash/zip-0032#specification-registered-key-derivation says:
> Let $\mathsf{ContextString}$ be a globally-unique non-empty sequence of at most 252 bytes that identifies the desired context.